src/functions/newsletter-submission.js

```diff
@@ -0,0 +1,51 @@
+const fetch = require("node-fetch");
+const { BUTTONDOWN_API_KEY } = process.env;
+
+const config = (userEmail) => {
+  return {
+    method: "POST",
+    headers: {
+      Authorization: `Token ${BUTTONDOWN_API_KEY}`,
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({ email: userEmail }),
+  };
+};
+
+exports.handler = async ({ body }) => {
+  const email = getEmail(body);
+
+  try {
+    const response = await fetch(
+      "https://api.buttondown.email/v1/subscribers",
+      config(email)
+    );
+
+    const data = await response.json();
+    if (!hasError(data)) {
+      return respondWith(400, "Could not sign user up to newsletter");
+    }
+    return respondWith(200, "Thanks for signing up!");
+  } catch (error) {
+    return respondWith(500, "Could not sign user up to newsletter");
```

Comment:
For this and the `400` above. Does the response from Buttondown include a statusCode? If so, we should use that, then fall back to custom codes.

Reply:
It does not include a status code for errors, just a messsage.